bin/fmt

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+set -eu
+
+go install -mod=readonly golang.org/x/tools/cmd/goimports
```

Comment:
i may not have my paths set up correctly, but this doesn't work out of the box in my env:
```bash
$ bin/fmt
bin/fmt: 13: bin/fmt: goimports: not found
```

rather than relying on (and modifying) the user's Go env, can you follow the pattern in `bin/lint` and other scripts to download the `goimports` binary into `./target/bin`?

i think it's fine to leave instructions in `BUILD.md` to enable folks to set up their env to use our blessed version of goimports, but this script should be less env-dependent.

Reply:
The `bin/lint` tool is downloaded differently, it downloads an install shell script and pipes it to sh which creates a binary in the current working directory.  Since we rely on go modules to install the correct version of this tool, I think we're stuck with whatever the behavior of `go install` is.

We could potentially run `mv $GOPATH/bin/goimports ${targetbin}/.goimports` after install, but I'm not sure if that's better.  

Does this work on your machine if you change the the `goimports` invocation to `$GOPATH/bin/goimports`?

Reply:
It's true `bin/lint` is a bit different given the pipe to shell pattern, but we're pretty consistent about wrapping tool downloading and execution this way:

```bash
$ grep "target/bin" bin/*
bin/docker:targetbin="$( cd $bindir/.. && pwd )"/target/bin
bin/goimports:targetbin="${rootdir}"/target/bin
bin/helm:targetbin="$( cd $bindir/.. && pwd )"/target/bin
bin/kind:targetbin="$( cd $bindir/.. && pwd )"/target/bin
bin/kubectl:targetbin="$( cd $bindir/.. && pwd )"/target/bin
bin/lint:targetbin="${rootdir}"/target/bin
bin/protoc:targetbin="$( cd $bindir/.. && pwd )"/target/bin
```

We can still leverage `go.mod` versioning while installing `goimports` into a specific location, by temporarily setting `GOBIN`. Recommend breaking out installing `goimports` from  `bin/fmt` into a new `bin/goimports` script:

`bin/goimports`:

```bash
#!/bin/bash

set -eu

cd "$(pwd -P)"

bindir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
rootdir="$( cd $bindir/.. && pwd )"
targetbin="${rootdir}"/target/bin

GOBIN=$targetbin go install -mod=readonly golang.org/x/tools/cmd/goimports

goimportsbin=${targetbin}/goimports

exec $goimportsbin "$@"
```

...and then `bin/fmt` can simply call `bin/goimports`:

```diff
diff --git a/bin/fmt b/bin/fmt
index b3e0eabf..b9ea373e 100755
--- a/bin/fmt
+++ b/bin/fmt
@@ -2,19 +2,17 @@

 set -eu

-go install -mod=readonly golang.org/x/tools/cmd/goimports
-
 # go list will exclude the vendor directory.  We use grep to exclude generated
 # code.
 dirs=$(go list -f \{\{.Dir\}\} ./... | grep -v controller/gen)
 # go list will list all subdirectories and goimports acts recursively.  This
 # results in certain files being reported multiple time.  Therefore, we must
 # dedup them.
-files=$(goimports -l $dirs | sort | uniq)
+files=$(bin/goimports -l $dirs | sort | uniq)

 if [ ! -z "$files" ]; then
   for file in $files; do
-    goimports -d "$file"
+    bin/goimports -d "$file"
   done
   exit 64
 fi
```